src/lib/schedule.c

```diff
@@ -343,7 +343,25 @@ void schedule_task_complete(struct task *task)
 	tracev_pipe("schedule_task_complete()");
 
 	spin_lock_irq(&sch->lock, flags);
-	task->state = TASK_STATE_COMPLETED;
+
+	/* Some high priority HW based IRQ handlers can reschedule tasks
+	 * immediately. i.e. before the task context can change task state
+	 * back to COMPLETED. Check here to make sure we dont clobber
+	 * task->state for regular non IRQ users.
+	 */
+	switch (task->state) {
+	case TASK_STATE_RUNNING:
+		task->state = TASK_STATE_COMPLETED;
+		break;
+	case TASK_STATE_QUEUED:
```

Comment:
@lgirdwood I will add case TASK_STATE_PENDING as well for the "do nothing" part and start the stress test.

Reply:
@ranj063 There might be the case.